fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -115,35 +113,31 @@ fn expand_function_arguments(
     fun: &Function,
     custom_enums: &HashMap<String, Property>,
     custom_structs: &HashMap<String, Property>,
-    strict_checking: bool,
 ) -> Result<(TokenStream, TokenStream), Error> {
     let n_inputs = fun.inputs.len();
-    let mut first_index = 0;
     // Check that we have the expected types
-    if strict_checking {
-        if n_inputs != 4 {
-            return Err(Error::MissingData(format!(
-                "Expected 4 inputs because `strict_checking` is true, found {}",
-                n_inputs
-            )));
-        }
-        let given_types: Vec<String> = fun.inputs[..3]
-            .iter()
-            .map(|x| x.type_field.clone())
-            .collect();
-        let expected = ["u64", "u64", "b256"];
-        if given_types != expected {
-            return Err(Error::InvalidType(format!(
-                "Expected the 3 first types to be {:?}, found {:?}",
-                expected, given_types
-            )));
-        };
-        // Ignore the first three arguments so we don't have to provide them when calling
-        // the contracts' methods
-        first_index = 3
+    if n_inputs != 4 {
+        return Err(Error::MissingData(format!(
+            "Expected exactly 4 inputs, found {}",
+            n_inputs
+        )));
     }
-    let mut args = Vec::with_capacity(n_inputs);
-    let mut call_args = Vec::with_capacity(n_inputs);
+    let given_types: Vec<String> = fun.inputs[..3]
+        .iter()
+        .map(|x| x.type_field.clone())
+        .collect();
+    let expected = ["u64", "u64", "b256"];
+    if given_types != expected {
+        return Err(Error::InvalidType(format!(
+            "Expected the 3 first types to be {:?}, found {:?}",
+            expected, given_types
+        )));
+    };
+    // Ignore the first three arguments so we don't have to provide them when calling
+    // the contracts' methods
+    let first_index = 3;
+    let mut args = Vec::with_capacity(1);
+    let mut call_args = Vec::with_capacity(1);
```

Comment:
We should add one more comment here to give some clarity around this logic. The for loop below: 

```rust
for (i, param) in fun.inputs[first_index..].iter().enumerate() {}
```

Only runs once under the assumption that we have a fixed input size of 4. A newcomer to the codebase would then ask, "why not just expand on the last element of `fun.inputs` then? Why the for loop?".

The only reason we're keeping this for loop is that we might, in the future, lift this restriction on the max number of inputs (4). So the for loop will make more sense then.